Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -72,6 +80,47 @@ const ContestCategories = ({
         updateBreadcrumb(category, categoriesFlat);
     }, [ possibleFilters, categoriesFlat, onCategoryClick, updateBreadcrumb ]);
 
+    const strategyFilterGroup = useMemo(
+        () => possibleFilters.filter(({ type }) => type === FilterType.Strategy),
+        [ possibleFilters ],
+    );
+
+    const newStrategyFilters = useCallback(
+        (id: string, node: ITreeItemType) => {
+            const filterProps = [];
+            if (currentStrategyFilters.map((csf) => csf.categoryId).includes(id)) {
+                currentStrategyFilters.filter((csf) => csf.categoryId !== id);
+            } else {
+                const strategyFiltersToAdd : IFilter[] = [];
+                node.allowedStrategyTypes.forEach((value) => {
+                    const currAllowedStrategyId = value.id.toString();
+
+                    const strategyToFind = strategyFilterGroup.find((x) => x.value === currAllowedStrategyId);
+
+                    if (!isNil(strategyToFind)) {
+                        strategyFiltersToAdd.push(strategyToFind);
+                    }
+                });
+
+                filterProps.push({
+                    categoryId: id,
+                    strategies: strategyFiltersToAdd,
+                });
+            }
+
+            const strategyFilters = filterProps.flatMap((s) => s.strategies)
+                .reduce((strategyFilter: IFilter[], currentStrategyFilter) => {
+                    strategyFilter.push(currentStrategyFilter);
+                    return strategyFilter;
+                }, []);
```

Comment:
Why is this `reduce` necessary? As I see it, the result is the same with or without it.